sklearn/neighbors/_regression.py

```diff
@@ -233,7 +233,7 @@ def predict(self, X):
         if weights is None:
             y_pred = np.mean(_y[neigh_ind], axis=1)
         else:
-            y_pred = np.empty((X.shape[0], _y.shape[1]), dtype=np.float64)
+            y_pred = np.empty((len(X), _y.shape[1]), dtype=np.float64)
```

Comment:
At this point, I think we can use `neigh_dist` to infer the number?

```suggestion
            y_pred = np.empty((neigh_dist.shape[0], _y.shape[1]), dtype=np.float64)
```

Reply:
Otherwise, we would tend to use `_num_samples` for array-like.

Reply:
Done!